agent/agent_endpoint.go

```diff
@@ -291,13 +302,13 @@ func (s *HTTPServer) AgentService(resp http.ResponseWriter, req *http.Request) (
 			ws.Add(svcState.WatchCh)
 
 			// Check ACLs.
-			rule, err := s.agent.resolveToken(token)
+			authz, err := s.agent.resolveToken(token)
```

Comment:
How come we don't get the authz from the call to `_, err := s.agent.resolveTokenAndDefaultMeta(token, &entMeta, nil)` above?

Reply:
This being a blocking query, potentially the token/roles/policies have changed since our original resolution. If its a short period of time when this fires then the resolver and identity will be cached anyways so it should be pretty quick to resolve a second time.